### ISSUE TITLE
Focusable and accessible native states can get out of sync

### ISSUE THREAD

**Issue by rozele** [microsoft/react-native-windows#10687]
### Problem Description

If only focusable or accessible props change in a particular batch of updates to a View, we will not apply the mitigation [here](https://github.com/microsoft/react-native-windows/blob/main/vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp#L610-L621) to ensure focusable and accessible states are kept in sync.

### Steps To Reproduce

1. Open the `view.tsx` example
2. Update the example with the code in [this gist](https://gist.github.com/rozele/00452accb6998e6ead60558a21f9eedd).
3. Play around with setting `pressable`, `focusable`, and `accessible` in various combinations.

Setting `pressable` to `false`, and then setting either `focusable` or `accessible` to `false` but not both puts the app in a bad state where narrator can get out of sync with keyboard focus.

See following bugs below:
1. IsTabStop not disabled after setting accessible to false
2. Is accessible not disabled after setting focusable to false 

### Expected Results

Based on #9840 and #10569, we should have the behavior described by the following pseudo-code:
```
if view is pressable:
  if view is not both focusable or accessible:
    view is not focusable and not accessible
  else:
    view is focusable and accessible
else:
  if view is focusable or accessible:
    view is focusable and accessible
  else:
    view is not focusable and not accessible
```

### CLI version

npx react-native --version

### Environment

```markdown
npx react-native info
```

### Snack, code example, screenshot, or link to a repository

https://user-images.githubusercontent.com/1106239/194372791-685a065d-f18f-4079-a2c8-c0d941cb49a2.mp4

**Comment by jonthysell** [comment]
@chiaramooney can take a look at this?
